lib/mastodon_user_processor.rb

```diff
@@ -282,7 +282,23 @@ def handle_force_url(content)
 
   def toot_content_to_post
     if toot.sensitive? && toot.spoiler_text.present?
-      "CW: #{toot.spoiler_text} … #{toot.url}"
+      if user.masto_cw_options == "cw_and_content"
+        if should_add_image_count?
+          "CW: #{toot.spoiler_text}\n\n#{toot.text_content}… #{toot.media_attachments.count} 🖼️"
+        else
+          "CW: #{toot.spoiler_text}\n\n#{toot.text_content}"
+        end
+      elsif user.masto_cw_options == "cw_only"
+        "CW: #{toot.spoiler_text} … #{toot.url}"
```

Comment:
I don't think you need to add the url here, you can use `self.force_toot_url = true`, and it will take care of everything else (including cutting the text if it's too long)

Reply:
I just copied what was in this block originally (line 285 of the file). I'm happy to make that change, but I want to be sure that it's the right thing to do, since it seems unrelated to the core idea of this pull request. (Or do you want to make that change, since it seems like you understand this mechanism better than I do?)

Reply:
Yeah, I think this is legacy code before I introduced the force option, it should behave exactly the same (and in the rare case the CW is too long, it will cut it automagically), I did a quick test on master and it should be fine, you can replace it by:
```
      self.force_toot_url = true
      "CW: #{toot.spoiler_text}"
```

Reply:
Also, I'm very aware that this function has gotten overcomplicated. Maybe it should be refactored to look more like this? (pseudocode)

```
tweet = toot.text_content
has_content = True
if toot is sensitive:
  if user.masto_cw_options == "cw_and_content":
    tweet = "CW: #{toot.spoiler_text}\n\n#{toot.text_content}"
  elsif user.masto_cw_options == "cw_only":
    tweet = "CW: #{toot.spoiler_text}"
    has_content = False
  elsif user.masto_cw_options == "content_only":
    tweet = toot.text_content
  else:
    raise "invalid masto_cw_options"

if should_add_image_count?
  tweet = "#{tweet}... #{toot.media_attachments.count} 🖼️"

return tweet
```